expression/scalar_function.go

```diff
@@ -70,8 +70,8 @@ func (sf *ScalarFunction) MarshalJSON() ([]byte, error) {
 	return []byte(fmt.Sprintf("\"%s\"", sf)), nil
 }
 
-// NewFunction creates a new scalar function or constant.
-func NewFunction(ctx sessionctx.Context, funcName string, retType *types.FieldType, args ...Expression) (Expression, error) {
+// newFunctionImpl creates a new scalar function or constant.
+func newFunctionImpl(fold bool, ctx sessionctx.Context, funcName string, retType *types.FieldType, args ...Expression) (Expression, error) {
```

Comment:
`ctx` is always the first argument of a function,
we'd better keep this style.

Reply:
Thank you for the detailed comment. I've updated the PR.